app/controllers/admin/purchases_controller.rb

```diff
@@ -42,11 +42,20 @@ def refund_taxes_only
   def resend_receipt
     if @purchase
       if params[:resend_receipt][:email_address].present?
-        @purchase.email = params[:resend_receipt][:email_address]
+        new_email = params[:resend_receipt][:email_address]
+        @purchase.email = new_email
         @purchase.save!
 
         user = User.alive.find_by(email: @purchase.email)
         @purchase.attach_to_user_and_card(user, nil, nil) if user
```

Comment:
This code can be DRY and re-used in both places.

Reassigning a purchase should only exist in one place of the code, whereas here it's twice.